firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -169,28 +171,26 @@ public void onFailure(@NonNull Exception e) {
     Task<InstallationTokenResult> installationAuthTokenTask =
         firebaseInstallationsApi.getToken(false);
 
-    Tasks.whenAllComplete(installationIdTask, installationAuthTokenTask)
+    Tasks.whenAllSuccess(installationIdTask, installationAuthTokenTask)
         .addOnSuccessListener(
             tasks -> {
               String fid = installationIdTask.getResult();
               InstallationTokenResult installationTokenResult =
                   installationAuthTokenTask.getResult();
               checkForUpdateExecutor.execute(
                   () -> {
-                    AppDistributionRelease latestRelease =
-                        getLatestReleaseFromClient(
-                            fid,
-                            firebaseApp.getOptions().getApplicationId(),
-                            firebaseApp.getOptions().getApiKey(),
-                            installationTokenResult.getToken());
-
-                    updateOnUiThread(
-                        new Runnable() {
-                          @Override
-                          public void run() {
-                            checkForUpdateTaskCompletionSource.setResult(latestRelease);
-                          }
-                        });
```

Comment:
Updated getLatestRelease to throw an exception instead of directly updating the checkForUpdate completion error. This way, we handle the error and success completion in one place.